src/util/Util.ts

```diff
@@ -286,3 +288,85 @@ export function timedPromise<T>(wrapped: Promise<T>, timeout: number, err?: Erro
 
     return deferred.promise;
 }
+
+/**
+ * Finds a larger same-version group of data members from a collection of
+ * members and, if `localMember` is from that group, return that.
+ * Otherwise return a random member from the group. If the same-version
+ * groups have the same size, return a member from the newer group
+ * (preferably the local one).
+ *
+ * Used for getting an SQL connection for executing SQL.
+ *
+ * @param members list of all members
+ * @param localMember the local member, null for client instance
+ * @throws IllegalStateError If there are more than 2 distinct member versions found
+ * @return the chosen member or null, if no data member is found
+ */
+export function memberOfLargerSameVersionGroup(members: MemberImpl[], localMember: MemberImpl | null): MemberImpl | null {
+    // The members should have at most 2 different version (ignoring the patch version).
+    // Find a random member from the larger same-version group.
+
+    let version0: MemberVersion | null = null, version1: MemberVersion | null = null;
+    let count0 = 0, count1 = 0;
+
+    const grossMajority = Math.trunc(members.length / 2); // Integer division like in Java side
+
+    for (const m of members) {
+        if (m.liteMember) {
+            continue;
+        }
+
+        const version = m.version;
+
+        let currentCount;
+
+        if (!version0 || version0.equals(version)) {
```

Comment:
Is this check correct? I see that we have `version0 == null || ...` in Java

Reply:
I think it is the same behaviour, but checking against null is better, changing..